src/archival/utils.rs

```diff
@@ -59,6 +74,50 @@ pub async fn is_row_exists(pool: &PgPool, row_id: i32) -> bool {
     }
 }
 
+pub async fn make_archival_network_request(
+    url: &str,
+    endpoint_url: &str,
+) -> Result<ArchivalResponse, ArchivalError> {
+    let settings = Settings::new().expect("Config settings are not configured properly");
+    let mut headers = header::HeaderMap::new();
+    headers.insert("Accept", "application/json".parse().unwrap());
+    headers.insert(
+        "Authorization",
+        format!(
+            "LOW {}:{}",
+            settings.wayback_machine_api.myaccesskey, settings.wayback_machine_api.mysecret
+        )
+        .parse()
+        .unwrap(),
+    );
+    headers.insert(
+        "Content-Type",
+        "application/x-www-form-urlencoded".parse().unwrap(),
+    );
+
+    let client = Client::builder()
+        .redirect(reqwest::redirect::Policy::none())
+        .build()
+        .unwrap();
+
+    let response = client
+        .post(endpoint_url)
+        .headers(headers)
+        .body(format!("url={}", url))
+        .send()
+        .await?;
+
+    // let response_status = response.status();
+    let response_text = response.text().await?;
+
+    if let Ok(res) = serde_json::from_str::<ArchivalResponse>(&response_text) {
+        return Ok(res);
+    }
+    Ok(ArchivalResponse::Html(ArchivalHtmlResponse {
+        html: response_text,
+    }))
```

Comment:
Definitely needs to be commented. I struggled to understand the point of this until I remembered reread the conversation